Holovibes/sources/thread/file_frame_read_worker.cc

```diff
@@ -269,17 +267,18 @@ void FileFrameReadWorker::enqueue_loop(size_t nb_frames_to_enqueue)
             {
             }
         }
+        input_queue_.load()->enqueue_multiple(gpu_file_frame_buffer_ + frames_enqueued * frame_size_,
+                                              setting<settings::BatchSize>(),
+                                              cudaMemcpyDeviceToDevice);
 
-        if (stop_requested_)
-            break;
-
-        input_queue_.load()->enqueue(gpu_file_frame_buffer_ + frames_enqueued * frame_size_, cudaMemcpyDeviceToDevice);
-
-        current_nb_frames_read_++;
-        processed_frames_++;
-        frames_enqueued++;
+        current_nb_frames_read_ += setting<settings::BatchSize>();
+        processed_frames_ += setting<settings::BatchSize>();
+        frames_enqueued += setting<settings::BatchSize>();
```

Comment:
You have no guarantee that `setting<settings::BatchSize>()` is a multiple of `nb_frames_to_enqueue`.
Instead you should enqueue: `std::min(setting<settings::BatchSize>(), nb_frames_to_enqueue - frames_enqueued)`.

In the futur it should be frame packet used here instead

Reply:
Resolved